### PR TITLE
support ubuntu path (~/snap/...)

### PR DESCRIPTION
**Pull request by saadtazi**
if linux, check if default firefox/ dir is present, if not, use ubuntu path.

### CODE REVIEW

File: lib/profile_finder.js [hunk 1]
```diff
@@ -16,6 +16,14 @@ function locateUserDirectory(platform) {
       break;
     case 'linux':
       userDir = path.join(process.env.HOME, '/.mozilla/firefox');
+      // fix: https://github.com/saadtazi/firefox-profile-js/issues/136
+      // check if directory is present, otherwise use ubuntu path: ~/snap/firefox/common/.mozilla/firefox/
+      if (!fs.existsSync(userDir)) {
```
Comment (jrom99) [inline lib/profile_finder.js:21]: I haven't tested my code yet, but it was along the same lines:

```js
snapDir = ...
dotDir = ...

xdgConfigDir = process.env.XDG_CONFIG_HOME || path.join(process.env.HOME, ".config")
xdgDir = path.join(xdgConfigDir, "mozilla/firefox")
if (fs.existsSync(dotDir)) { userDir = dotDir }
else if (fs.existsSync(snapDir)) { userDir = snapDir }
else { userDir = xdgDir }
```

**Comment by jrom99** [comment]
Mentioning the issue for easier access: https://github.com/saadtazi/firefox-profile-js/issues/136

File: lib/profile_finder.js (hunk 1 above)
Comment (saadtazi) [reply lib/profile_finder.js:21]: Ah, I wasn't sure what the XDG_CONFIG_HOME was used for, thank you!

**Comment by jrom99** [comment]
This may be a dumb question, and actually unrelated to the issue, but is there any reason in particular to use `Class.prototype` instead of JS class syntax?

File: lib/profile_finder.js (hunk 1 above)
Comment (saadtazi) [reply lib/profile_finder.js]: Pushed some changes, it's now closer to the logic you proposed.

**Comment by saadtazi** [comment]
> This may be a dumb question, and totally unrelated to the issue, but is there any reason in particular to use `Class.prototype` instead of JS class syntax?

good question: `firefox-profile` package was created in october 2013 (more than 10 years ago 👴 ), at that time, `class` and `const/let` didn't exist in nodejs. `class` was introduced in nodejs 4.0.0 (in 2015).

I do not use it myself anymore, so I didn't take the time to update the code (nor the test/CI tools)

**Comment by jrom99** [comment]
On second thought, I think `process.env` does not listen to snap or flatpak overrides for firefox.

In this case, I think the best behavior would be to assume that we can be dealing with a sandboxed or unsandboxed firefox install, where the normal home can still be a valid location for profile data, since both [snap](https://github.com/canonical/firefox-snap/blob/d60bc91f31c89708529194142864dcfea15f5f75/snapcraft.yaml#L34C7-L34C32) and [flatpak](https://hg.mozilla.org/mozilla-central/file/tip/taskcluster/docker/firefox-flatpak/runme.sh#l160) seem to allow it.

So the code could be something like:

```js
function locateLinuxDirectory() {
  userHomeDir = process.env.HOME;

  // look for a valid sandboxed profile dir
  homes = [
     path.join(userHomeDir, "snap/firefox/common"),
     path.join(userHomeDir, ".var/app/org.mozilla.firefox")
  ]
  for (home of homes) {
     dotDir = path.join(home, ".mozilla/firefox")
     xdgDir = path.join(home, ".config/mozilla/firefox")
     if (fs.existsSync(dotDir) {
       return dotDir
     } else if (fs.existsSync(xdgDir)) {
       return xdgDir
     }
  }

  // look in the next probable locations
  dotDir = path.join(userHomeDir, ".mozilla/firefox")
  xdgConfigDir = process.env.XDG_CONFIG_HOME || path.join(userHomedir, ".config")
  xdgDir = path.join(xdgConfigDir, "mozilla/firefox")
  return fs.existsSync(dotDir) ? dotDir : xdgDir;
}
```

**Comment by saadtazi** [comment]
sorry for the delay, I didn't have access to a computer... pushed some changes...

**Comment by jrom99** [comment]
I think the logic is still inverted for global installs. The dot dir should be preferred if it exists, but it should not be the default directory.

Also, the XDG config dir doesn't need a leading dot.

**Review by jrom99 — APPROVED** [review]

**Pull request merged**